File: snippets/java/src/main/java/foundations/actions/OrderWorkflow.java
package foundations.actions;

import dev.restate.sdk.annotation.Shared;
import dev.restate.sdk.annotation.Workflow;

@Workflow
public class OrderWorkflow {
  @Workflow
  public void run(Order order) {
    // Implementation
  }

  @Shared
  public String getStatus() {
    return "pending";
  }
}

File: snippets/python/src/foundations/actions/order_workflow.py
import restate
from pydantic import BaseModel
from restate import WorkflowContext, WorkflowSharedContext

order_workflow = restate.Workflow("OrderWorkflow")


class Order(BaseModel):
    pass


@order_workflow.main()
async def run(ctx: WorkflowContext, order: Order) -> None:
    pass


@order_workflow.handler()
async def get_status(ctx: WorkflowSharedContext) -> str:
    return "pending"
